app/src/main/java/com/pajato/android/gamechat/database/AccountManager.java

```diff
@@ -308,8 +309,20 @@ public void signIn(final Activity activity, final String login, final String pas
 
     /** Sign in using the given User account. */
     public void signIn(final Context context, final String provider, final String accountName) {
-        // Invoke the sign in activity to kick off a Firebase auth event.
-        Intent intent = new Intent(context, SignInActivity.class);
+        // Get an instance of AuthUI based on the default app, and build an intent.
```

Comment:
It appears that your intent builder is identical in at least two signIn() methods but the setup is inlined.  Looks like an opportunity for a procedural abstraction, `getIntentBuilder()`